utils/wasm-builder/src/prerequisites.rs

```diff
@@ -138,6 +138,11 @@ fn check_wasm_toolchain_installed(
 	build_cmd.env_remove("CARGO_TARGET_DIR");
 	run_cmd.env_remove("CARGO_TARGET_DIR");
 
+	// Make sure the host's flags aren't used here, e.g. if an alternative linker is specified
+	// in the RUSTFLAGS then the check we do here will break unless we clear these.
+	build_cmd.env("CARGO_ENCODED_RUSTFLAGS", "");
+	run_cmd.env("CARGO_ENCODED_RUSTFLAGS", "");
```

Comment:
```suggestion
	build_cmd.env("CARGO_ENCODED_RUSTFLAGS", "");
	run_cmd.env("CARGO_ENCODED_RUSTFLAGS", "");
	build_cmd.env("RUSTFLAGS", "");
	run_cmd.env("RUSTFLAGS", "");
```

Reply:
I might be totally wrong here since I didn't look into this too closely, but I *think* for any new versions of `cargo` it might only expose `CARGO_ENCODED_RUSTFLAGS` instead of `RUSTFLAGS` (basically even if you set `RUSTFLAGS` that will get converted to `CARGO_ENCODED_RUSTFLAGS` and stripped out). Nevertheless it probably shouldn't hurt to also clear `RUSTFLAGS` too!

Reply:
As we execute here cargo "manually" as you do it as human, it will also pick up `RUSTFLAGS`. However, it will first check if `CARGO_ENCODED_RUSTFLAGS` exists and then fall back.

Reply:
Yeah, you're right. I was mostly looking at it from the perspective of this code running through a normal `cargo build` in `polkadot` where it isn't being ran directly.